dashboard/cron.py

```diff
@@ -298,13 +305,16 @@ def update_with_bq_access(self):
             # Because we're pulling all the data down into one query we need to manipulate it a little bit
             # Make a copy of the access dataframe
             resource_df = resource_access_df.copy(deep=True)
-            # Drop out the columns user and access time from resource data frame
-            resource_df.drop(["user_id", "access_time"], axis=1, inplace=True)
+            # Drop out the columns user course_id and access time from resource data frame
+            resource_df = resource_df.drop(["user_id", "course_id", "access_time"], axis=1)
             # Drop out the duplicates
-            resource_df.drop_duplicates(["resource_id", "course_id"], inplace=True)
+            resource_df = resource_df.drop_duplicates(["resource_id",])
+
+            # Set a dual index for upsert
+            resource_df = resource_df.set_index(["resource_id", "course_id"])
```

Comment:
Yep, as noted by @pushyamig, this is throwing an error. Your data model no longer has a `course_id` on `Resource`, so I tried removing `"course_id"` and things seemed to work. Maybe this line and the comment above are an artifact of a previous iteration.

Also, how have you been testing the updating? Do you have a test course set up?

Reply:
Thanks Sam, I'll check and see if I just missed pushing an update for this.

I was testing this by manually manipulating the database in the local destination database rather than waiting for the source to change.

I'd delete some rows from resource_access to make sure the new ones get inserted and change some values in resource. I think otherwise we'll have to use a test course like you suggest and make changes over a few days.